TheAmazingAudioEngine/AEAudioUnitFileStreamer.m

```diff
@@ -0,0 +1,395 @@
+        
+        // Set the file to play
+        AudioUnit au = AEAudioUnitChannelGetAudioUnit(self);
+        checkResultForError(AudioUnitSetProperty(au, kAudioUnitProperty_ScheduledFileIDs, kAudioUnitScope_Global, 0, &_audioUnitFile, sizeof(_audioUnitFile)), "AudioUnitSetProperty(kAudioUnitProperty_ScheduledFileIDs)", "Failed setting audio unit file IDs", NO, result, error)
+        
+        // Get audio description
+        size = sizeof(_audioDescription);
+        checkResultForError(AudioFileGetProperty(_audioUnitFile, kAudioFilePropertyDataFormat, &size, &_audioDescription), "AudioFileGetProperty(kAudioFilePropertyDataFormat)", "Failed getting audio file data format", NO, result, error)
+        
+        // Get file length
+        UInt64 fileLengthInFrames = 0;
+        size = sizeof(fileLengthInFrames);
+        checkResultForError(ExtAudioFileGetProperty(audioFileRef, kExtAudioFileProperty_FileLengthFrames, &size, &fileLengthInFrames), "ExtAudioFileGetProperty(kExtAudioFileProperty_AudioFile)", "Failed getting file length", NO, result, error)
+        
+        _lengthInFrames = (UInt32)fileLengthInFrames;
+        _url = url;
+        
+        return(YES);
+    }
+    else {
+        *error = [NSError errorWithDomain:NSOSStatusErrorDomain code:result userInfo:[NSDictionary dictionaryWithObject:@"URL failed to open" forKey:NSLocalizedDescriptionKey]];
+        return(NO);
+    }
+}
+
+- (void)dealloc {
+    
+    if ( _audioUnitFile ) {
+        AudioFileClose(_audioUnitFile);
+    }
+    
+    _completionBlock = nil;
+    _audioController = nil;
+}
+
+// MARK: Audio functions
+
+- (void) play
+{
+    self.channelIsPlaying = true;
+}
+- (void) pause
+{
+    self.channelIsPlaying = false;
+}
+- (void) stop
+{
+    self.channelIsPlaying = NO;
+    _playhead = 0;
+    NSError* error;
+    [self setupPlayRegion:&error];
+    if(error)
+        NSLog(@"Error on playback stopped callback: %@",error.description);
+}
+
+- (void) mute
+{
+    self.channelIsMuted = true;
+}
+
+- (void) unmute
+{
+    self.channelIsMuted = false;
+}
+
+// MARK: Getters and Setters
+-(NSTimeInterval)duration {
+    return (double)_lengthInFrames / (double)_audioDescription.mSampleRate;
+}
+
+-(NSTimeInterval)currentTime {
+    if (_lengthInFrames == 0) return 0.0;
+    else return ((double)_playhead / (double)_lengthInFrames) * [self duration];
+}
+
+-(void)setCurrentTime:(NSTimeInterval)currentTime {
+    if (_lengthInFrames == 0) return;
+    _playhead = (int32_t)((currentTime / [self duration]) * _lengthInFrames) % _lengthInFrames;
+    
+    // Reload the audio region
+    NSError* error;
+    [self setupPlayRegion:&error];
+    if(error) {
+        NSLog(@"Error trying to change the playhead time. %@",error.description);
+    }
+}
```

Comment:
Same thoughts on this one. Create channel, add it, set current time, play it. However, I think we should move the _lengthInFrames calculation so that you don't have to follow that order if you don't want to.